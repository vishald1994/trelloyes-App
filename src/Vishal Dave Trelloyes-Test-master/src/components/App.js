import React from 'react';
import '../styles/App.css';
import List from './List';

class App extends React.Component {
  render() {
    const store = this.props.store
    const lists = store.lists.map(list => {
      const cards = list.cardIds.map(id => store.allCards[id]);

      return (
        <List
          key={list.id}
          header={list.header}
          cards = {cards}
        />
      )
    });

    return (
      <main className="App">
        <header className="App-header">
          <h1>
            Trelloyes!
          </h1>
        </header>
        <div className="App-list">
          {lists}
        </div>
      </main>
    )
  }
}

export default App;