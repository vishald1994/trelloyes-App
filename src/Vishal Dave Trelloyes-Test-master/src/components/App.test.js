import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import STORE from '../js/store';
import renderer from 'react-test-renderer';

/**
 * App Component Smoke Test
 */
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App store={STORE} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

/**
 * App Component Snapshot Test
 */

it ('renders this UI as expected', () => {
  const app = renderer.create(<App store={STORE}/>).toJSON();
  expect(app).toMatchSnapshot();
})
