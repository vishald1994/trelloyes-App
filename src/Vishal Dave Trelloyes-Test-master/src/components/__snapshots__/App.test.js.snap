// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders this UI as expected 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Trelloyes!
    </h1>
  </header>
  <div
    className="App-list"
  >
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          First list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Twelfth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </section>
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          Second list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </section>
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          Third list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Twelfth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </section>
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2>
          Fourth list
        </h2>
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Twelfth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </section>
  </div>
</main>
`;
